Draft new charter and goals for v49/v50 and beyond (#1065)
unicode-org/message-format-wg: docs/goals.md

# Goals and Non-Goals

This document contains the charter for the MessageFormat Working Group (MFWG)
and informs decisions about the scope and priority of its efforts.

## Charter

A **_dynamic message string_** is a string whose content changes due to the value of or insertion
of some data value or values.

The _Unicode MessageFormat Standard_ is an industry standard for the representation
of localizable _dynamic message strings_.

The MessageFormat Working Group maintains and extends the Unicode MessageFormat Standard,
provides documentation;
encourages implementation, including the development of tools and best practices;
manages default and Unicode-defined function sets;
and provides for interoperability with other standards.

> The MessageFormat Working Group is a working group the CLDR-TC.

Reviewer: ```suggestion
The MessageFormat Working Group is a working group of the CLDR-TC.
```

## Goals

- Encourage adoption of Unicode MessageFormat as measured by developing [messageformat.unicode.org]
  and other appropriate materials to include a user guide, MF1 migration guide,
  implementation support, and usage materials while maintaining a high bar 

>   as a model for other Unicode websites.

Author: > Do we really want to "provide ... participation in workshops, conferences, ..."? That sounds like we'd be funding such activities. Is that intentional?

Nope, just noodling.

> - Encourage adoption of Unicode MessageFormat as measured by developing [messageformat.unicode.org]
>   and other appropriate materials to include a user guide, MF1 migration guide,
>   implementation support, and usage materials while maintaining a high bar 
>   as a model for other Unicode websites.

Author: ```suggestion
- Maintain and develop the [messageformat.unicode.org] site,
  maintaining a high bar (as a model for other Unicode documentation),
  including at least 
  a user guide,
  and a migration guide from other formats (including ICU MessageFormat).
- Encourage adoption of Unicode MessageFormat by:
  providing implementation support materials;
  creating tests;
  linking to success stories, tools, and implementations;
  supporting or hosting workshops or supporting presentations by members;
  and supporting adoption by encapsulating standards (such as ECMA-402, MessageResource, etc.)
```

> - Support a determination there are at least ### additional interoperable implementations.

Reviewer: This seems more like a measure of how well we've succeeded at the preceding goal.
```suggestion
```

Author: The is a strawman for an implementation goal. Yes, it's a measure of how well our documentation has worked, but I envision a slightly more muscular approach to platforms teams (especially members)

- Support migration and adoption by moving all functions and options necessary to match the features of

>   ICU MessageFormat ("MF1") to Stable.

Author: Discussed in 2025-05-19 call. This needs to be changed.

>   ICU MessageFormat ("MF1") to Stable.

Author: Fixed

and provides for interoperability with other standards.

The MessageFormat Working Group is a working group of the CLDR-TC.

## Goals

- Maintain and develop the [messageformat.unicode.org] site,
  maintaining a high bar (as a model for other Unicode documentation),
  including at least 
  a user guide,
  and a migration guide from other formats (including ICU MessageFormat).
- Encourage adoption of Unicode MessageFormat by:
  providing implementation support materials;
  creating tests;
  linking to success stories, tools, and implementations;
  supporting or hosting workshops or supporting presentations by members;
  and supporting adoption by encapsulating standards (such as ECMA-402, MessageResource, etc.)
- Support a determination there are at least ### additional interoperable implementations.
- Support migration and adoption by moving all functions and options necessary to match the features of
  ICU MessageFormat ("MF1") to Stable.
- Support migration and adoption by expanding the default function set to support
  additional use cases.
- Develop a machine-readable function description format or syntax to support the needs of
  implementations, including localization tools.

> 

Reviewer: As follow-up from discussions on last week's call, I think we should add the following:
```suggestion
- Define a standard vocabulary for expression attributes and message properties/metadata,
  to enable better interoperation between translation tools and platforms.
- Incubate and support working groups or interest groups
  that promote adoption of Unicode MessageFormat,
  such as the proposed working group to develop a standard message resource format,
  i.e. a new file format for bundles or collections of messages.

```

## Goals

- Maintain and develop the [messageformat.unicode.org] site,
  maintaining a high bar (as a model for other Unicode documentation),
  including at least 
  a user guide,
  and a migration guide from other formats (including ICU MessageFormat).
- Encourage adoption of Unicode MessageFormat by:
  providing implementation support materials;
  creating tests;
  linking to success stories, tools, and implementations;
  supporting or hosting workshops or supporting presentations by members;
  and supporting adoption by encapsulating standards (such as ECMA-402, MessageResource, etc.)
- Support migration and adoption by publishing as Stable in the default function set
  the functions and options needed to provide compatibility with ICU MessageFormat ("MF1").
  Note that certain features of MF1 have been deliberately excluded from this goal
  because they are in conflict with the design goals of Unicode MessageFormat,
  because they are specialized features unique to MF1 (which ICU is free to implement),
  or because they might be superseded by newer specifications:
  - ChoiceFormat (an anti-pattern)
  - Date/time picture strings (an anti-pattern)

>   - Date/time and number skeletons (specialized functionality, potentially superseded by semantic skeletons)

Reviewer 2: ```suggestion
  - Date/time and number skeletons (potentially superseded by semantic skeletons)
```
I think the above covers enough of a reason to exclude for now.

  - Nested selection or partial message selection (an anti-pattern)
  - `spellout` and `duration` formats (specialized functionality)

>   - `ordinal` _formatting_ (specialized functionality, incompetely specified in CLDR)

Reviewer 2: ```suggestion
  - `ordinal` _formatting_ (specialized functionality)
```
Ordinal formatting is accomplished by having the ordinal categories, and inflections (in where needed). Both are _defined_ in CLDR.
What is lacking is not the definition, but rather that some languages lack data.

## Deliverables

> The original deliverables were:

Author: It's temporary while we work through the current charter.
